service/task.go

```diff
@@ -840,67 +791,238 @@ func (uis *UIServer) taskModify(w http.ResponseWriter, r *http.Request) {
+			http.Error(w, "not found", http.StatusNotFound)
 			return
 		}
+
+		ctx, cancel := context.WithCancel(r.Context())
+		defer cancel()
+
+		data.Data = make(chan apimodels.LogMessage)
+		go func() {
+			defer close(data.Data)
+			for _, line := range testLog.Lines {
+				if ctx.Err() != nil {
+					return
+				}
+				data.Data <- apimodels.LogMessage{
+					Type:     apimodels.TaskLogPrefix,
+					Severity: apimodels.LogInfoPrefix,
+					Version:  evergreen.LogmessageCurrentVersion,
+					Message:  line,
+				}
+			}
+		}()
+	}
+
+	template := "task_log.html"
+	if raw {
+		template = "task_log_raw.html"
+		uis.renderText.Stream(w, http.StatusOK, data, "base", template)
 	} else {
-		taskID := vars["task_id"]
-		testName := vars["test_name"]
-		taskExecutionsAsString := vars["task_execution"]
-		taskExec, err = strconv.Atoi(taskExecutionsAsString)
-		if err != nil {
-			http.Error(w, "task execution num must be an int", http.StatusBadRequest)
-			return
+		uis.render.WriteResponse(w, http.StatusOK, data, "base", template)
+	}
+}
+
+func (uis *UIServer) getTestResults(w http.ResponseWriter, r *http.Request, projCtx projectContext, uiTask *uiTaskData, execution int) {
+	ctx := r.Context()
+	var err error
+
+	uiTask.TestResults = []uiTestResult{}
+	execTasks := []task.Task{}
+	execTaskIDs := []string{}
+	if uiTask.DisplayOnly {
+		for _, t := range projCtx.Task.ExecutionTasks {
+			var et *task.Task
+			if uiTask.Archived {
+				et, err = task.FindOneOldNoMergeByIdAndExecution(t, execution)
+			} else {
+				et, err = task.FindOneId(t)
+			}
+			if err != nil {
+				uis.LoggedError(w, r, http.StatusInternalServerError, err)
+				return
+			}
+			if et == nil {
+				grip.Error(message.Fields{
+					"message": "execution task not found",
+					"task":    t,
+					"parent":  projCtx.Task.Id,
+				})
+				continue
+			}
+			execTasks = append(execTasks, *et)
+			execTaskIDs = append(execTaskIDs, t)
 		}
 
-		testLog, err = model.FindOneTestLog(testName, taskID, taskExec)
+		execTasks, err = task.MergeTestResultsBulk(execTasks, nil)
 		if err != nil {
 			uis.LoggedError(w, r, http.StatusInternalServerError, err)
 			return
 		}
+		for i, execTask := range execTasks {
+			uiTask.ExecutionTasks = append(uiTask.ExecutionTasks, uiExecTask{
+				Id:        execTaskIDs[i],
+				Name:      execTask.DisplayName,
+				TimeTaken: execTask.TimeTaken,
+				Status:    execTask.ResultStatus(),
+			})
+		}
 	}
 
-	if testLog == nil {
-		http.Error(w, "not found", http.StatusNotFound)
-		return
-	}
+	// Check cedar test results first.
+	success := true
+	if uiTask.DisplayOnly {
+		allResults := []uiTestResult{}
+		for i, execTask := range execTasks {
+			opts := apimodels.GetCedarTestResultsOptions{
+				BaseURL:   uis.Settings.Cedar.BaseURL,
+				TaskID:    execTask.Id,
+				Execution: execTask.Execution,
+			}
+			testResults, err := apimodels.GetCedarTestResults(ctx, opts)
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
```

Comment:
Do we have to log this every time when none of the projects are using this so far? It'll produce a lot of spurious errors until every single project is moved to using Cedar test results and no older test results are accessed.

As an idea, can we make Cedar return 404s when it doesn't find any test results matching the task ID/test name and reduce this to `grip.Debug()` when we get 404s?

Reply:
I changed it to a warning, it does the same thing for task logs and hasn't been an issue.

Reply:
It still seems to be an error.